docs/zedctr.md

```diff
@@ -0,0 +1,96 @@
+|xen tools|`zededa/xen-tools`|[pkg/xen-tools](../pkg/xen-tools)|`/`|Xen utilities||
+|dnsmasg|`zededa/dnsmasq`|[pkg/dnsmasq](../pkg/dnsmasq)|`/usr/sbin/dnsmasq` to `/opt/zededa/bin/dnsmasq`|local DNS server|Unclear why we need a local build|
+|strongswan|`zededa/strongswan`|[pkg/strongswan](../pkg/strongswan)|`/`|IPSec VPN|Dockerfile has [comment](../pkg/strongswan/Dockerfile#L63-L66), stating that we build custom because of libressl conflict with curl-dev. Should be investigated if this remains an issue|
+|gpt tools|`zededa/gpt-tools`|[pkg/gpt-tools](../pkg/gpt-tools)|`/`|Utilities for working with GPT partitioned filesystems. See [here](./UPSTREAMING.md#gpt-tools) for more details on what is in this package|
+|watchdog|`zededa/watchdog`|[pkg/watchdog](../pkg/watchdog)|`/usr/sbin` to `/usr/sbin`|[watchdog]() daemon to watch for userspace processes and, if they are not providing proper heartbeats, restart the system|Unclear why we need to restart the system vs restarting zombie processes, a la systemd/runit/supervisord/etc.|
+
+In addition, a few local installations are provided as necessary, defined in the [Dockerfile](../pkg/zedctr/Dockerfile). They are:
+
+* `libfdt` via `apk` for `arm64` only
+* symlink `/usr/lib/libpcap.so` to `libpcap.so.1`
+* generate a host ssh key, configure `/etc/ssh/sshd_config` to permit root login via ssh, and change the ssh root key. It is unclear why we are doing this for several reasons:
+    * We are generating the root password at package build time, storing secrets in a static OCI image. 
+    * We are generating the ssh host key at package build time, causing all hosts to have the same ssh host key.
+    * We are allowing root login via ssh
+* copy the local [pkg/zedctr/rootfs](../pkg/zedctr/rootfs/) into the image at `/`
+* Set the `CMD /init.sh`
+
+#### lisp-install
+
+lisp files are copied as follows:
+
+```
+COPY --from=lisp /lisp/lisp-ztr /opt/zededa/bin/
+COPY --from=lisp /lisp /opt/zededa/lisp/
+COPY --from=lisp /usr/bin/pydoc /usr/bin/smtpd.py /usr/bin/python* /usr/bin/
+COPY --from=lisp /usr/lib/libpython* /usr/lib/libffi.so* /usr/lib/
+COPY --from=lisp /usr/lib/python2.7 /usr/lib/python2.7/
+```
+
+#### zedctr/rootfs
+
+[zedctr/rootfs](../pkg/zedctr/rootfs) is a small group of files that are installed in the root to `/` of the `zedctr` container. This document will not list every file, but will give the general purposes.
+
+* [blink.sh](../pkg/zedctr/rootfs/blink.sh) - appears to be a simple utility to exercise `/dev/sda` by copying 22 blocks of 4M to `/dev/null`, a process that should take about 200ms, then sleeps 200ms, repeating 3 times, then sleeping 1200ms, then repeating. Its usage is unknown.
+* [sos.sh](../pkg/zedctr/rootfs/sos.sh) - similar to blink, but an SOS pattern instead. Its usage is unknown.
```

Comment:
This is a test program which we can delete.

Reply:
Same